hw/ip/prim/data/prim_mubi_pkg.sv.tpl

```diff
@@ -0,0 +1,156 @@
+<%
+## This Mako template can be rendered using
+##
+##     hw/ip/prim/util/generate_prim_mubi_pkg.py hw/ip/prim/data/prim_mubi_pkg.sv.tpl > \
+##                                               hw/ip/prim/rtl/prim_mubi_pkg.sv
+##
+%>\
+
+package prim_mubi_pkg;
+
+% for n in range(1, n_max_nibbles + 1):
+<%
+  nbits = n * 4
+  hi_val = ''
+  lo_val = ''
+  for k in range(1,n+1):
+    hi_val = ('5' if (k % 2) else 'A') + hi_val
+    lo_val = ('A' if (k % 2) else '5') + lo_val
+%>\
+  //////////////////////////////////////////////
+  // ${nbits} Bit Multibit Type and Functions //
+  //////////////////////////////////////////////
+
+  parameter int MuBi${nbits}Width = ${nbits};
+  typedef enum logic [MuBi${nbits}Width-1:0] {
+    MuBi${nbits}Hi = ${nbits}'h${hi_val}, // enabled
+    MuBi${nbits}Lo = ${nbits}'h${lo_val}  // disabled
+  } mubi${nbits}_e;
+
+  // Set the multibit value to signal "enabled".
+  function automatic mubi${nbits}_e mubi${nbits}_set_hi();
+    return MuBi${nbits}Hi;
+  endfunction : mubi${nbits}_set_hi
+
+  // Set the multibit value to signal "disabled".
+  function automatic mubi${nbits}_e mubi${nbits}_set_lo();
+    return MuBi${nbits}Lo;
+  endfunction : mubi${nbits}_set_lo
+
+  // Test whether the multibit value signals an "enabled" condition.
+  // The strict version of this function requires
+  // the multibit value to equal Hi.
+  function automatic logic mubi${nbits}_tst_hi_strict(mubi${nbits}_e val);
+    return MuBi${nbits}Hi == val;
+  endfunction : mubi${nbits}_tst_hi_strict
+
+  // Test whether the multibit value signals a "disabled" condition.
+  // The strict version of this function requires
+  // the multibit value to equal Lo.
+  function automatic logic mubi${nbits}_tst_lo_strict(mubi${nbits}_e val);
+    return MuBi${nbits}Lo == val;
+  endfunction : mubi${nbits}_tst_lo_strict
+
+  // Test whether the multibit value signals an "enabled" condition.
+  // The loose version of this function interprets all
+  // values other than Lo as "enabled".
+  function automatic logic mubi${nbits}_tst_hi_loose(mubi${nbits}_e val);
+    return MuBi${nbits}Lo != val;
+  endfunction : mubi${nbits}_tst_hi_loose
+
+  // Test whether the multibit value signals a "disabled" condition.
+  // The loose version of this function interprets all
+  // values other than Hi as "disabled".
+  function automatic logic mubi${nbits}_tst_lo_loose(mubi${nbits}_e val);
+    return MuBi${nbits}Hi != val;
+  endfunction : mubi${nbits}_tst_lo_loose
+
+
+  // Performs a logical OR operation between two multibit values.
+  // This treats "act" as logical 1, and all other values are
+  // treated as 0. Truth table:
+  //
+  // A    | B    | OUT
+  //------+------+-----
+  // !act | !act | !act
+  // act  | !act | act
+  // !act | act  | act
+  // act  | act  | act
+  //
+  function automatic mubi${nbits}_e mubi${nbits}_or(mubi${nbits}_e a, mubi${nbits}_e b, mubi${nbits}_e act);
+    logic [${nbits}-1:0] a_in, b_in, act_in, out;
```

Comment:
Throughout:
```suggestion
    logic [${nbits - 1}:0] a_in, b_in, act_in, out;
```

`bit [4-1:0]` just seems weird in the generated code.

Reply:
agreed